backend/persistent/datastores/csv.js

```diff
@@ -0,0 +1,98 @@
+const alasql = require('alasql');
+import fetch from 'node-fetch';
+import papa from 'papaparse';
+import {type} from 'ramda';
+
+import {parseSQL} from '../../parse';
+
+// define type error thrown by this connector
+export function CSVError(url, errors) {
+      this.name = 'CSVError';
+      this.message = 'Failed to parse CSV file ' + url;
+
+      if (Error.captureStackTrace) {
+          Error.captureStackTrace(this, CSVError);
+      } else {
+          this.stack = new Error(this.message).stack;
+      }
+
+      this.url = url;
+      this.errors = errors;
+
+      if (errors && errors[0] && errors[0].message) {
+          this.message = errors[0].message;
+      }
+}
+CSVError.prototype = Object.create(Error.prototype);
+CSVError.prototype.constructor = CSVError;
+
+// the data parsed from CSV files is stored here
+const connectionData = {};
+function getData(connection) {
+    return connectionData[connection.database];
+}
+function putData(connection, data) {
+    connectionData[connection.database] = data;
+}
+
+export function connect(connection) {
```

Comment:
If the url is bad, how will this be presented to the user on the Settings UI?  will fetch throw an error and this will be gracefully passed to the UI on its own?

Reply:
Same behaviour for all the connectors, i.e.: fetch throws, `routes.js` catches the exception and replies to the connection request with a 500 status and the body `{error: message: error.message}}`.